### DISCUSSION TITLE
Chat filter based on the number of special/accented characters in the message.

### DISCUSSION THREAD

**Discussion by mtang125** [runelite/runelite#15606] [Ideas]
This is related to these 2 posts I found to try to combat bot spammer:
https://github.com/runelite/runelite/issues/13097
https://github.com/runelite/runelite/issues/14350

Could we add a feature to filter chat messages based on the number of special/accented characters that occur within the message? This can be a player defined threshold, and if the number of special characters is greater than the threshold, the message will not appear in the chat.

Just another idea in the endless battle against bots.

**Comment by Nightfirecat** [comment]
You can always write a regex which fits that criteria.

**Comment by mtang125** [comment]
Ah true. I played around with it for it bit a found this regex works pretty well. This will filter on messages that contain 5 or more accented characters. 
```
^([^À-ÿ]*[À-ÿ]){5,}[^À-ÿ]*$
```

Although, it seems to have some conflicts when you have `Strip accents` enabled. It seems to be filtering legitimate messages containing whitespace for some reason. And the same messages don't get filtered with `Strip accents` disabled. Not sure why or how to fix that.
